### DISCUSSION TITLE
How to restart the edgedb server from a shell script running in the edgedb Docker container?

### DISCUSSION THREAD

**Discussion by elliotwaite** [geldata/gel#6650] [Q&A]
If I'm running the edgedb Docker container, how do I restart the server from a shell script running inside the container?

I tried the following, but ran into the error shown below:

`$ edgedb instance link --dsn='edgedb://edgedb:password@localhost:5656?tls_security=insecure' localhost-edgedb`
> `Successfully linked to remote instance. To connect run:
edgedb -I localhost-edgedb`

`$ edgedb instance restart -I localhost-edgedb`
> `edgedb error: error reading instance info: "/root/.local/share/edgedb/data/localhost-edgedb/instance_info.json": No such file or directory (os error 2)`

Am I doing something wrong?

For context, I'm trying to write a shell script that will restart the server whenever the TLS files change (the files pointed to by `EDGEDB_SERVER_TLS_CERT_FILE` and `EDGEDB_SERVER_TLS_KEY_FILE`), and they are changed externally by a separate certbot container that has a shared volume with the edgedb container. Is there a more correct way to accomplish what I'm trying to do?

**UPDATE:**
I just realized that the EdgeDB server already handles watching for TLS file changes, and automatically loads in the new certificates when they do. So the original context of my question no longer applies, but I'm still curious how I would restart the local server if needed.

**Comment by elprans** [comment]
Remote instances, a.k.a ones that you `instance link` cannot be restarted by the CLI as it has no control over the remote process, you would need to `instance restart` from the same logical host where you ran `instance create`/`project init`.  We need a better error message of course.

**Reply by elliotwaite** [reply]
Ah, okay. However, since I'm using the EdgeDB Docker container, I never run the `instance create`/`project init` commands to create the instance. Are these commands (or something similar) run automatically by the container when it starts up? I tried calling `edgedb instance list` from a shell within the container, but it replied with "No instances found" (I tried running the command as both the `root` user and as the `edgedb` user to see if that would make a difference, but I got the "No instances found" message in both cases). What would I do in the case of using the EdgeDB Docker container where I never run the `instance create`/`project init` commands?
